### DISCUSSION TITLE
Print out confusion matrix in training and testing

### DISCUSSION THREAD

**Discussion by nicozhou7** [open-mmlab/mmaction2#644] [Q&A] [answered]
I have a question, how do I modify to code in order to print out the confusion matrix in training and testing?

**Comment by innerlee** [comment, marked as answer]
You can implement it as a kind of evaluation metric. Similar to how mean_class_accuracy is done https://github.com/open-mmlab/mmaction2/blob/master/mmaction/datasets/base.py#L202

**Comment by irvingzhang0512** [comment]
If you try to show other metrics during training(along with top1_acc, top5_acc, loss, etc), you should modify head. For recognizers, you should modify [BaseHead](https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/heads/base.py), add keys for dict `losses`
